nupm/registry.nu

```diff
@@ -0,0 +1,178 @@
+# Registry management for nupm
+
+use utils/dirs.nu [nupm-home-prompt REGISTRY_FILENAME]
+use utils/log.nu throw-error
+
+# Manage nupm registires
+@example "List all configured registries" { nupm registry }
+export def main []: nothing -> table {
+    list
+}
+
+# List all configured registries
+@example "List all registries with details" { nupm registry list }
+export def list []: nothing -> table {
+    $env.NUPM_REGISTRIES | transpose name url | sort-by name
+}
+
+
+def registry-names [] { list | get name }
+# Show detailed information about a specific registry
+# returning a list of package names, type, and version
+@example "Show registry information" { nupm registry describe nupm }
+export def describe [
+    registry: string@registry-names
+]: nothing -> table {
+    use utils/dirs.nu cache-dir
+
+    if not ($registry in $env.NUPM_REGISTRIES) {
+        throw-error $"Registry '($registry)' not found"
+    }
+
+    let registry_url = $env.NUPM_REGISTRIES | get $registry
+    let registry_cache_dir = cache-dir --ensure | path join $registry
```

Comment:
I think you're missing the `registry` subdirectory here. It now dumps the files into `$env.NUPM_CACHE/<registry-name>` instead of `$env.NUPM_CACHE/registry/<registry-name>`.

Reply:
nice catch, I think there should be some followup tests on this